Assets/Shopify/UIToolkit/Theme Controllers/SingleProductThemeController.cs

```diff
@@ -8,7 +8,16 @@
 
     [AddComponentMenu("Shopify/Theme Controllers/Single Product Theme Controller")]
     public class SingleProductThemeController : ThemeControllerBase {
-        public ISingleProductTheme Theme;
+        public new ISingleProductTheme Theme { 
+            get { 
+                return base.Theme as ISingleProductTheme; 
+            }
+
+            set {
+                base.Theme = value;
+            } 
+        }
```

Comment:
this is a neat trick for hiding the theme from the base controller and always getting the more specific interface

Reply:
Neat trick! I had no idea you could use the `new` keyword like that.